java/client/src/main/java/glide/api/models/Script.java

```diff
@@ -0,0 +1,40 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import static glide.ffi.resolvers.ScriptResolver.dropScript;
+import static glide.ffi.resolvers.ScriptResolver.storeScript;
+
+/**
+ * A wrapper for a Script object for {@link
+ * glide.api.commands.GenericBaseCommands#invokeScript(Script)} As long as this object is not
+ * closed, the script's code is saved in memory, and can be resent to the server.
+ */
+public class Script implements AutoCloseable {
+
+    /** hash string representing the code */
+    private String hash;
+
+    /**
+     * Wraps around creating a Script object from <code>code</code>.
+     *
+     * @param code To execute with a ScriptInvoke call
+     */
+    public Script(String code) {
+        this.hash = storeScript(code);
+    }
+
+    /**
+     * Retrieve the stored hash
+     *
+     * @return hash
+     */
+    public String getHash() {
```

Comment:
When using getHash they seem to be using the ffi function getHash (I do not understand fully why) but is this also something to be consider maybe it's to make sure the backend has that value? [PY Example](https://github.com/aws/glide-for-redis/commit/2ea5d6caddd3db02baa51e46a5f37836cf6ff61e#diff-f0f69819afe466c0f3e8afd4ea2fd31ca4f451373fa8ed1351f48473a1c39633R1274)

Reply:
I saw that.  I will bring it up with the Amazon team _why_ they are storing the hash on the rust side rather than on the python/node side.  I don't understand their reasoning here.